src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,54 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.KoraException;
+
+/**
+ * Parser class process user input and generates command.
+ */
+public class Parser {
+
+    /**
+     * Class constructor of Parser.
+     */
+    public Parser() {
+    }
+
+    /**
+     * Processes user input and returns a command based on data processed.
+     * @param userInput String input from user.
+     * @return Command to be executed.
+     * @throws KoraException For invalid input.
+     */
+    public static Command parse(String userInput) throws KoraException {
+        String line = "------------------------------" + "\n";
+        String[] userInputArray = userInput.split("/");
+        Command command;
+        try {
+            if (userInputArray[0].contains("bye")) {
+                command = new ByeCommand();
+            } else if (userInputArray[0].contains("list")) {
+                command = new ListCommand();
+            } else if (userInputArray[0].contains("unmark")) {
+                command = new UnmarkCommand(userInputArray);
+            } else if (userInputArray[0].contains("mark")) {
+                command = new MarkCommand(userInputArray);
+            } else if (userInputArray[0].contains("deadline")) {
+                command = new DeadlineCommand(userInputArray);
+            } else if (userInputArray[0].contains("event")) {
+                command = new EventCommand(userInputArray);
+            } else if (userInputArray[0].contains("todo")) {
+                command = new ToDoCommand(userInputArray);
+            } else if (userInputArray[0].contains("delete")) {
+                command = new DeleteCommand(userInputArray);
+            } else if (userInputArray[0].contains("find")) {
+                command = new FindCommand(userInputArray);
```

Comment:
The `userInputArray[0].contains("find")` will return true as long as it has a substring "find" in it. In this case, "asdfacads**find**adfaefedcef" will also be considered as a valid command. Similar for the others. What do you think?

Reply:
Good point.. I will take note. Thanks!